Palindrome_permutation.py

```diff
@@ -1,25 +1,34 @@
-def is_permutation_of_palindrome(x):
-    # remove spaces and convert strings to lowercase
-    x = x.replace(" ", "").lower()
-
-    char_count = {}
-
-    for i in x:
-        if i.isalpha():
-            if i in char_count:
-                char_count[i] += 1
-            else:
-                char_count[i] = 1
-
-    odd_count = 0
-
-    for j in char_count.values():
-        if j % 2 != 0:
-            odd_count += 1
-
-    return odd_count <= 1
-
-a = str(input())
-print(is_permutation_of_palindrome(a))
+# In this problem we check if it's possible to obtain a mirrored string by simply rearranging the letters. "
+# If so, return - True, if not, return - False
+class Solution:
+    def __init__(self, str1):
+        self.str1 = str1
+
+    def is_permutation_of_palindrome(self):
+        # remove spaces and convert strings to lowercase
+        self.str1 = self.str1.replace(" ", "").lower()
+        char_count = {}
+
+        # Check if the character is an alphabet letter.
+        for i in self.str1:
+            if i.isalpha():
+                if i in char_count:
+                    char_count[i] += 1
+                else:
+                    char_count[i] = 1
+
+        odd_count = 0
+        
+        # Check if the count is odd.
+        for j in char_count.values():
+            if j % 2 != 0:
+                odd_count += 1
+
+        return odd_count <= 1

```

Comment:
Instead of so much code this can be done using `Counter` in python

```
cleaned_input = input.string().lower() // note I don't remove all spaces since it doesn't matter and save time by doing on less operation on the string. Remember string manipulation are always expensive since it loops on each character one by one.
counter = Counter(cleaned_input)
return sum(val % 2 for val in counter.values()) <= 1
```